Mathlib/Order/BooleanAlgebra.lean

```diff
@@ -326,6 +326,15 @@ theorem le_sdiff_iff : x ≤ y \ x ↔ x = ⊥ :=
   ⟨fun h => disjoint_self.1 (disjoint_sdiff_self_right.mono_right h), fun h => h.le.trans bot_le⟩
 #align le_sdiff_iff le_sdiff_iff
 
+@[simp] lemma sdiff_eq_right : x \ y = y ↔ x = ⊥ ∧ y = ⊥ := by
+  refine ⟨fun h ↦ ?_, ?_⟩
+  · obtain ⟨h, rfl⟩ := disjoint_sdiff_self_left.eq_iff.1 h
+    simpa using h
+  · rintro ⟨rfl, rfl⟩
+    exact bot_sdiff
```

Comment:
Is there a `himp` analog?

Reply:
Yes, but only in boolean algebras because we don't "generalized coboolean algebras" or whatever that is called